tests/backup_test.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/bash
+
+include './src/backup.sh'
+include './src/kwlib.sh'
+include './tests/utils.sh'
+
+function setUp()
+{
+  KW_DATA_DIR="$SHUNIT_TMPDIR/data"
+  decompress_path="$SHUNIT_TMPDIR/tmp-kw-backup"
+
+  mkdir -p "$KW_DATA_DIR"
+  mkdir -p "$decompress_path"
+
+  cp -r tests/samples/configs "$KW_DATA_DIR"
+  cp -r tests/samples/pomodoro_data "$KW_DATA_DIR/pomodoro"
+  cp -r tests/samples/statistics "$KW_DATA_DIR"
+}
+
+function test_create_backup()
+{
+  local output
+  local filepath
+  declare -a expected_files
+  declare -a tar_files
+
+  mapfile -t expected_files < <(for f in $(find "$KW_DATA_DIR" -type f | sort -d); do
+    str_remove_prefix "$f" "$KW_DATA_DIR/"
+  done)
```

Comment:
uouuuu... mapfile :)
Add a brief comment that explains what these set of line goals are.